wsi/drm/drmdisplay.cpp

```diff
@@ -419,30 +407,150 @@ bool DrmDisplay::SetBroadcastRGB(const char *range_property) {
   return true;
 }
 
-void DrmDisplay::SetHDCPState(HWCContentProtection state,
-                              HWCContentType content_type) {
-  desired_protection_support_ = state;
-  content_type_ = content_type;
-  if (desired_protection_support_ == current_protection_support_)
-    return;
+DrmDisplay::DrmHDCPThread::DrmHDCPThread(uint32_t gpu_fd, uint32_t connector,
+                                         uint32_t hdcp_id, uint32_t hdcp_cp_id)
+    : HWCThread(-8, "HDCPThread"),
+      gpu_fd_(gpu_fd),
+      connector_(connector),
+      hdcp_id_(hdcp_id),
+      hdcp_cp_id_(hdcp_cp_id) {
+}
+
+bool DrmDisplay::DrmHDCPThread::CheckHDCPStatus() {
+  int val = -1;
+  uint32_t i;
+  ScopedDrmObjectPropertyPtr cprops(drmModeObjectGetProperties(
+      gpu_fd_, connector_, DRM_MODE_OBJECT_CONNECTOR));
+  for (i = 0; i < cprops->count_props; i++) {
+    if (cprops->props[i] == hdcp_id_) {
+      val = cprops->prop_values[i];
+      break;
+    }
+  }
+
+  if ((current_state_ && val == 2) || (!current_state_ && val == 0))
+    return true;
 
-  if (hdcp_id_prop_ <= 0) {
+  return false;
+}
+
+bool DrmDisplay::DrmHDCPThread::DrmConnectorSetupHDCP() {
```

Comment:
Instead of doing this in HWC, HDCP Deamon may have similar logic,
I think we need to sync with the VPG CP team about this.
We should avoid duplidated logic

Reply:
Yes, I initially thought so too. But the compositor needs to be an independent entity. I believe contacting HDCP daemon would require a binder call? Then that would incur overhead. 
Also, HDCP daemon is available only on Android. What about the other OSes/distros we support?

Reply:
@harishkrupo  @js0701  Do you consider this issue resolved?